### PR TITLE
[Merged by Bors] - Remove fallback support from eth1 service

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

N/A

## Proposed Changes

With https://github.com/sigp/lighthouse/pull/3214 we made it such that you can either have 1 auth endpoint or multiple non auth endpoints. Now that we are post merge on all networks (testnets and mainnet), we cannot progress a chain without a dedicated auth execution layer connection so there is no point in having a non-auth eth1-endpoint for syncing deposit cache. 

This code removes all fallback related code in the eth1 service. We still keep the single non-auth endpoint since it's useful for testing.

## Additional Info

This removes all eth1 fallback related metrics that were relevant for the monitoring service, so we might need to change the api upstream.

### CODE REVIEW

File: beacon_node/src/lib.rs [hunk 1]
```diff
@@ -117,7 +117,7 @@ impl<E: EthSpec> ProductionBeaconNode<E> {
             info!(
                 log,
                 "Block production enabled";
-                "endpoints" => format!("{:?}", &client_config.eth1.endpoints),
+                "endpoints" => format!("{:?}", &client_config.eth1.endpoint),
```
Comment (michaelsproul) [inline beacon_node/src/lib.rs]:
```suggestion
                "endpoint" => format!("{:?}", &client_config.eth1.endpoint),
```

File: beacon_node/eth1/src/service.rs [hunk 2]
```diff
@@ -255,11 +137,11 @@ async fn get_remote_head_and_new_block_ranges(
...
             "action" => "trying fallback"
         );
-        return Err(SingleEndpointError::EndpointError(EndpointError::FarBehind));
+        return Err(Error::EndpointError(EndpointError::FarBehind));
     }
 
     let handle_remote_not_synced = |e| {
-        if let SingleEndpointError::RemoteNotSynced { .. } = e {
+        if let Error::RemoteNotSynced { .. } = e {
             warn!(
```
Comment (michaelsproul) [inline beacon_node/eth1/src/service.rs:141]: there's an `"action"` key in this warn log that says `"trying fallbacks"`. we should delete it

File: beacon_node/eth1/src/service.rs [hunk 3]
```diff
@@ -53,127 +50,12 @@ const CACHE_FACTOR: u64 = 2;
...
-                    reset_endpoint_state(endpoint).await;
-                }
-            }
-        }
-    }
-}
-
 /// Returns `Ok` if the endpoint is usable, i.e. is reachable and has a correct network id and
 /// chain id. Otherwise it returns `Err`.
 async fn endpoint_state(
```
Comment (michaelsproul) [inline beacon_node/eth1/src/service.rs:61]: There's another instance of `"trying fallbacks"` in this function. We should delete all of them.

**Review by michaelsproul — CHANGES_REQUESTED** [review]
This is awesome!

Looks good on the whole, I've just requested a few minor changes to the logs.

I've also marked this PR as backwards incompat, as it will break `--eth1-endpoints` for users on Gnosis chain. Given the Gnosis merge should happen soon after we release this, I think it's OK to just note the breakage in the release notes.

**Comment by pawanjay176** [comment]
To account for removed fallback related metrics that were sent over the monitoring service endpoint, this PR sets default values for all monitoring metrics that aren't present as lighthouse metrics in 9a71f41

Currently, the affected metrics would be `sync_eth1_fallback_configured` and `eth1_sync_fallback_connected`
cc @Buttaa

**Review by michaelsproul — APPROVED** [review]
LGTM!

**Pull request closed without merging**